src/app/search/SearchFilter.tsx

```diff
@@ -122,15 +120,51 @@ class SearchFilter extends React.Component<Props, State> {
       } else {
         // Bulk tagging
         const itemInfoService = await getItemInfoSource(this.props.account!);
+        const selectedTagString = bulkItemTags.find(
+          (tagInfo) => tagInfo.type && tagInfo.type === selectedTag
+        ).label;
         const tagItems = this.getStoresService()
           .getAllItems()
           .filter((i) => i.taggable && this.props.searchFilter(i));
+        // existing tags are later passed to buttonEffect so the notif button knows what to revert
+        const previousState = tagItems.map((item) => {
+          return { item, setTag: item.dimInfo.tag as TagValue | 'clear' | 'lock' | 'unlock' };
+        });
         await itemInfoService.bulkSaveByKeys(
           tagItems.map((item) => ({
             key: item.id,
             tag: selectedTag === 'clear' ? undefined : (selectedTag as TagValue)
           }))
         );
+        showNotification({
+          type: 'success',
+          duration: 30000,
+          title: t('Header.BulkTag'),
+          body: [
+            t(selectedTagString === 'Tags.ClearTag' ? 'Filter.BulkClear' : 'Filter.BulkTag', {
+              count: tagItems.length,
+              tag: t(selectedTagString)
+            }),
+            <NotificationButton
+              key="bulktaggingundobutton"
```

Comment:
Huh, I wouldn't expect you to need a key here.

Reply:
it's sibling content to the string above it, so TS demanded it be keyed since it's part of an array

i could not massage the `body` key into a string+JSX concatenation, tried a bunch, if you have a solution i'd love it

Reply:
You can wrap it in a fragment `<></>`

Reply:
weird i tried a couple permutations but never found the right combination of braces and parentheses. new editor helped solve this :)